callbacks/src/main/java/swt/DefaultOpenPopupCallback.java

```diff
@@ -0,0 +1,107 @@
+import com.teamdev.jxbrowser.browser.event.BrowserClosed;
+import com.teamdev.jxbrowser.browser.event.TitleChanged;
+import com.teamdev.jxbrowser.browser.event.UpdateBoundsRequested;
+import com.teamdev.jxbrowser.ui.Rect;
+import com.teamdev.jxbrowser.view.swt.BrowserView;
+import org.eclipse.swt.SWTException;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Widget;
+
+/**
+ * The default {@link OpenPopupCallback} implementation for the Swing UI toolkit that creates and shows a new window
+ * with the embedded popup browser.
+ */
+public final class DefaultOpenPopupCallback implements OpenPopupCallback {
+
+    private static final int DEFAULT_POPUP_WIDTH = 800;
+    private static final int DEFAULT_POPUP_HEIGHT = 600;
+
+    @Override
+    public Response on(Params params) {
+        Browser browser = params.popupBrowser();
+        try {
+            Display display = Display.getDefault();
+            display.asyncExec(() -> {
+                Shell shell = new Shell(display);
+                shell.setLayout(new FillLayout());
+                BrowserView view = BrowserView.newInstance(shell, browser);
+                updateBounds(shell, view, params.initialBounds());
+
+                shell.addDisposeListener(event -> {
+                    if (!browser.isClosed()) {
+                        asyncExec(shell, browser::close);
+                    }
+                });
+
+                browser.on(TitleChanged.class, event ->
+                        asyncExec(shell, () -> shell.setText(event.title())));
+
+                browser.on(BrowserClosed.class, event ->
+                        asyncExec(shell, shell::dispose));
+
+                browser.on(UpdateBoundsRequested.class, event ->
+                        asyncExec(shell, () -> updateBounds(shell, view, event.bounds())));
+
+                view.setVisible(true);
+                shell.pack();
+                shell.open();
+
+                while (!shell.isDisposed()) {
+                    if (!display.readAndDispatch()) {
+                        display.sleep();
+                    }
+                }
+            });
+        } catch (SWTException ignore) {
+            Response.proceed();
+        }
+        return Response.proceed();
+    }
+
+    private static void updateBounds(Shell shell, BrowserView view, Rect bounds) {
+        shell.setLocation(bounds.x(), bounds.y());
+        if (bounds.size().isEmpty()) {
+            view.setSize(DEFAULT_POPUP_WIDTH, DEFAULT_POPUP_HEIGHT);
+        } else {
+            view.setSize(bounds.width(), bounds.height());
+        }
+    }
+
+    private static void asyncExec(Widget widget, Runnable doRun) {
+        try {
+            widget.getDisplay().asyncExec(() -> {
+                if (!widget.isDisposed()) {
+                    doRun.run();
+                }
+            });
+        } catch (SWTException ignore) {
```

Comment:
This is a strange code to be in the example. If I were a client I'd read it as "we can throw some exceptions inside, just ignore them".  It smells. I know that it's not far from the truth and I regularly saw the exceptions "*** is already disposed" during the browser closing. 

I think this detail is redundant in this example, let's go with the standard `Display.asyncExec` everywhere.

Reply:
We mustn't call methods of the disposed `Shell`. I have removed the `try-catch` block.